src/iop/liquify.c

```diff
@@ -405,7 +403,7 @@ static void node_gc(dt_iop_liquify_params_t *p)
       k++;
   }
   //  invalidate all nodes beyond the last moved one
-  for(int k=last+1; k<MAX_NODES; k++)
+  for(k=last+1; k<MAX_NODES; k++)
```

Comment:
I'd prefer keeping the loop iterator limited to the loop. If you do not want to use shadowed k just use another letter for the index.